docs/pages/examples/contract-write-dynamic.en-US.mdx

```diff
@@ -0,0 +1,357 @@
+
+We will need to:
+
+1. Add the hook
+2. Pass in the transaction hash (`data.hash`) as a parameter to the hook
+2. Add loading state to the button when the transaction is pending.
+3. Add a success state for when the transaction is successful.
+
+```tsx {5,20,22-24,38-48}
+import * as React from 'react'
+import { 
+  usePrepareContractWrite, 
+  useContractWrite,
+  useWaitForTransaction
+} from 'wagmi'
+import { useDebounce } from './useDebounce'
+
+export function MintNFTForm() {
+  const [tokenId, setTokenId] = React.useState('')
+  const debouncedTokenId = useDebounce(tokenId)
+
+  const { config } = usePrepareContractWrite({
+    addressOrName: '0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2',
+    contractInterface: ['function mint(uint32 tokenId)'],
+    functionName: 'mint',
+    args: [parseInt(debouncedTokenId)],
+    enabled: Boolean(debouncedTokenId)
+  })
+  const { data, write } = useContractWrite(config)
+
+  const { isLoading, isSuccess } = useWaitForTransaction({
+    hash: data?.hash,
+  })
+
+  return (
+    <form onSubmit={(e) => {
+      e.preventDefault()
+      write?.()
+    }}>
+      <label for="tokenId">Token ID</label>
+      <input 
+        id="tokenId" 
+        onChange={e => setTokenId(e.target.value)}
+        placeholder="420"
+        value={tokenId}
+      />
+      <button disabled={!write || isLoading}>
+        {isLoading ? 'Minting...' : 'Mint'}
+      </button>
+      {isSuccess && (
+        <div>
+          Successfully minted your NFT!
+          <div>
+            <a href={`https://etherscan.io/tx/${data?.hash}`}>Etherscan</a>
+          </div>
+        </div>
+      )}
+    </form>
+  )
+}
+```
+
+## Step 8: Add To App
+
+Import the `MintNFTForm` component and display it when the account is connected.
+
+```tsx {3,12}
+import { useAccount, useConnect, useDisconnect } from 'wagmi'
+
+import { MintNFTForm } from './MintNFTForm'
+
+export function App() {
+  const { isConnected } = useAccount()
+
+  if (isConnected) {
+    return (
+      <div>
+        {/* Account content goes here */}
+        <MintNFTForm />
+      </div>
+    )
+  }
+
+  return <div>{/* Connect wallet content goes here */}</div>
+}
+```
+
+## Bonus Point: Add some error handling
+
+Now let's provide some feedback to the user if the prepare hook fails, or if the write hook fails.
+
+```tsx {15-16,24,53-55}
+import * as React from 'react'
+import { 
+  usePrepareContractWrite, 
+  useContractWrite,
+  useWaitForTransaction
+} from 'wagmi'
+import { useDebounce } from './useDebounce'
+
+export function MintNFTForm() {
+  const [tokenId, setTokenId] = React.useState('')
+  const debouncedTokenId = useDebounce(tokenId)
+
+  const { 
+    config,
+    error: prepareError,
+    isError: isPrepareError,
+  } = usePrepareContractWrite({
+    addressOrName: '0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2',
+    contractInterface: ['function mint(uint32 tokenId)'],
```

Comment:
Could be nice to use JS object format for the ABI so it works well when #941 is merged in.
```suggestion
    contractInterface: [{
      name: 'mint',
      type: 'function',
      stateMutability: 'nonpayable',
      inputs: [{ internalType: 'uint256', name: 'tokenId', type: 'uint256' }],
      outputs: [],
    }],
```

Reply:
Oh of course!